osquery/core/init.cpp

```diff
@@ -570,11 +573,25 @@ void Initializer::start() const {
     DatabasePlugin::setAllowOpen(true);
     // A daemon must always have R/W access to the database.
     DatabasePlugin::setRequireWrite(tool_ == ToolType::DAEMON);
-    if (!DatabasePlugin::initPlugin()) {
-      LOG(ERROR) << RLOG(1629) << binary_
-                 << " initialize failed: Could not initialize database";
-      auto retcode = (isWorker()) ? EXIT_CATASTROPHIC : EXIT_FAILURE;
-      requestShutdown(retcode);
+
+    for (size_t i = 1; i <= kDatabaseMaxRetryCount; i++) {
+      auto ok = DatabasePlugin::initPlugin();
+      if (ok) {
+        break;
+      }
+
+      if (i == kDatabaseMaxRetryCount) {
+        LOG(ERROR) << RLOG(1629) << binary_
+                   << " initialize failed: Could not initialize database";
+        auto retcode = (isWorker()) ? EXIT_CATASTROPHIC : EXIT_FAILURE;
+        requestShutdown(retcode);
+      }
+
+      LOG(WARNING) << RLOG(1629) << binary_
```

Comment:
made some changes:
* changed inti status check to: `if (DatabasePlugin::initPlugin().ok()) { ..`
* reduced  the delay to 200 msec, increased the number of attempts to 25 (should give almost 5 seconds of waiting in total)
* removed warning and info logging

Let me know if it works.

Thank you!